dotcom-rendering/src/layouts/lib/decideAdSlots.tsx

```diff
@@ -0,0 +1,152 @@
+import { Hide } from '@guardian/source-react-components';
+import { AdSlot } from '../../components/AdSlot.web';
+import { MAX_FRONTS_BANNER_ADS } from '../../lib/commercial-constants';
+import { frontsBannerAdCollections } from '../../lib/frontsBannerAbTestAdPositions';
+import { frontsBannerExcludedCollections } from '../../lib/frontsBannerExclusions';
+import { getMerchHighPosition } from '../../lib/getAdPositions';
+
+export const decideMerchHighAndMobileAdSlots = (
+	renderAds: boolean,
+	index: number,
+	collectionCount: number,
+	isPaidContent: boolean | undefined,
+	mobileAdPositions: (number | undefined)[],
+	hasPageSkin: boolean,
+	showBannerAds?: boolean,
+) => {
+	if (!renderAds) return null;
+
+	const minContainers = isPaidContent ? 1 : 2;
+	const shouldInsertMerchHighSlot =
+		collectionCount > minContainers &&
+		index === getMerchHighPosition(collectionCount);
+
+	if (shouldInsertMerchHighSlot) {
+		return showBannerAds ? (
+			<Hide from="desktop">
+				<AdSlot
+					data-print-layout="hide"
+					position="merchandising-high"
+					hasPageskin={hasPageSkin}
+				/>
+			</Hide>
+		) : (
+			<AdSlot
+				data-print-layout="hide"
+				position="merchandising-high"
+				hasPageskin={hasPageSkin}
+			/>
+		);
+	} else if (mobileAdPositions.includes(index)) {
+		return (
+			<Hide from="tablet">
+				<AdSlot
+					index={mobileAdPositions.indexOf(index)}
+					data-print-layout="hide"
+					position="mobile-front"
+				/>
+			</Hide>
+		);
+	}
+
+	return null;
+};
+
+/**
+ * Renders a fronts-banner ad when in the fronts banner AB test.
+ * Only applies to network fronts on desktop screens and wider.
+ */
+export const decideFrontsBannerAdSlot = (
+	renderAds: boolean,
+	hasPageSkin: boolean,
+	numBannerAdsInserted: React.MutableRefObject<number>,
```

Comment:
Do we need this mutable ref? I think a plain variable in the scope of this new module would do the job?

Reply:
Yeah I think you're right. Let's test this afterwards but I see no reason why we need a ref over a variable here.

Reply:
[useRef](https://react.dev/reference/react/useRef) should be equivalent

Reply:
refs don't force a re-render worth checking that you are not relying on it to update state used by the render.

https://react.dev/reference/react/useRef